internal/pkg/deploy/task.go

```diff
@@ -22,3 +27,23 @@ type CreateTaskResourcesInput struct {
 
 	AdditionalTags map[string]string
 }
+
+// TaskStackInfo contains essential information about a Copilot task stack
+type TaskStackInfo struct {
+	StackName string
+	App       string
+	Env       string
+
+	RoleARN string
+}
+
+// TaskName returns the name of the one-off task. This is the same as the value of the
+// copilot-task tag.
+func (t TaskStackInfo) TaskName() string {
```

Comment:
Could we update the `NameForTask` in `stack` pkg to
```go
type TaskStackName string

func (t TaskStackName) TaskName() string {
	return strings.SplitN(t.StackName, "-", 2)[1]
}

// NameForTask returns the stack name for a task.
func NameForTask(task string) TaskStackName {
	return fmt.Sprintf("task-%s", task)
}
```

Benefit is that we won't get lost when we redefine the task stack name format.